src/utils/client.ts

```diff
@@ -328,6 +332,7 @@ export default class Client {
   submitAuthInfo({ authId, authPhone = '', authEmail = '' }: any) {
     const { userKey } = initInfo();
     initState.authId = authId;
+    if (setAuthId) setAuthId(authId);
```

Comment:
we have to wait for the stytch or firebase call to finish and authId be returned - so there is still a noticeable lag between the page loading and navigating to the correct step

Reply:
I don't think we can optimistically make an update - because the pure presence of stytch tokens in URl etc doesn't mean they are correct & that the auth call will succeed. So, we'd run the risk of letting a user into an auth gate section when their auth fails

I think I may need to detect if auth is going to happen, and delay rendering the entire form so we can go straight to the correct step